include/zephyr/sys/util.h

```diff
@@ -685,6 +685,16 @@ char *utf8_trunc(char *utf8_str);
  */
 char *utf8_lcpy(char *dst, const char *src, size_t n);
 
+/**
+ * @brief Checks if the given string @p str is UTF-8 encoded.
+ *
+ * @param str Target string
+ * @param len The length of string @p str
+ *
+ * @return true if @p str is UTF-8 encoded, or false otherwise.
+ */
+bool utf8_check(const char *str, size_t maxlen);
```

Comment:
Either prefix the function name with `is_` (same as `is_power_of_two` and `is_null_no_warn`) or change the return type to `int` and return `0` on success.

Reply:
I would like to ask that it will not be changed, because I hope to keep the prefix of this toolset consistent, so that it will be more convenient to prompt in the editor. Look this
![图片](https://github.com/user-attachments/assets/398596d0-22de-4060-8a39-90084d8ec236)

Reply:
Guess this is a matter of taste, but for me, mixing up the two return styles (int vs bool) is way worse (harder to read, easier to make errors) than having to look up a function name (harder to write). Code gets read more often than written.

I was under the impression that Zephyr had some guidelines on this topic, but I can not find it anymore. Can anyone help us out here?

Reply:
`utf8_is_valid()` would be some compromise.

Reply:
`utf8_is_valid` it's good name :)

Reply:
Sounds great to me, thanks!

Reply:
Please also update the commit message accordingly.